src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -889,10 +910,36 @@ void AndroidDeviceControllerWrapper::OnICDRegistrationInfoRequired()
 void AndroidDeviceControllerWrapper::OnICDRegistrationComplete(chip::NodeId icdNodeId, uint32_t icdCounter)
 {
     chip::DeviceLayer::StackUnlock unlock;
+
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::app::ICDClientInfo clientInfo;
+    clientInfo.peer_node         = ScopedNodeId(icdNodeId, Controller()->GetFabricIndex());
+    clientInfo.monitored_subject = mAutoCommissioner.GetCommissioningParameters().GetICDMonitoredSubject().Value();
+    clientInfo.start_icd_counter = icdCounter;
+
+    err = mICDClientStorage.SetKey(clientInfo, mAutoCommissioner.GetCommissioningParameters().GetICDSymmetricKey().Value());
+    if (err == CHIP_NO_ERROR)
+    {
+        err = mICDClientStorage.StoreEntry(clientInfo);
+    }
+
+    if (err == CHIP_NO_ERROR)
+    {
+        ChipLogProgress(Controller, "Saved ICD Symmetric key for " ChipLogFormatX64, ChipLogValueX64(icdNodeId));
+    }
+    else
+    {
+        mICDClientStorage.RemoveKey(clientInfo);
```

Comment:
if failed, print error, then early return? currently you are calling onICDRegistrationComplete even thought it fails.

Reply:
The current DevicePairingDelegate implementation does not consider stopping it. Even in other error cases, the error code is transmitted only to the upper layer.

Reply:
ack

Reply:
I will add errorCode in this input parameter.

Reply:
then we just need to inject error inside onICDRegistrationComplete if any for this function. thanks

Reply:
I already added in onICDRegistrationComplete.
https://github.com/project-chip/connectedhomeip/pull/31193/files#diff-9e353b12e06979cac624339aaf9d8b9c7da07cd4ebe98157dee888cdd297002cR1625